src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/hotheap/HighHeapUsageYoungGenRca.java

```diff
@@ -132,6 +208,22 @@ protected void remove(SlidingWindowData e) {
           MEM_TYPE.getField(), OLD_GEN.toString(), MetricsDB.MAX);
       if (!Double.isNaN(oldGenHeapUsed)) {
         promotionRateDeque.next(new SlidingWindowData(currTimeStamp, oldGenHeapUsed / CONVERT_BYTES_TO_MEGABYTES));
+        if (oldGenHeapUsed > maxOldGen) {
+          maxOldGen = oldGenHeapUsed;
+        }
+        double promoted = oldGenHeapUsed - prevOldGen;
+        if (promoted > 0) {
+          youngGenPromotedMB += promoted / CONVERT_BYTES_TO_MEGABYTES;
+        } else { // full GC must have occurred if there's less data in the old gen
+          double garbageReclaimedPct = (prevOldGen - oldGenHeapUsed) / (youngGenPromotedMB * CONVERT_BYTES_TO_MEGABYTES);
```

Comment:
`HighHeapUsageOldGenRca` has a `minOldGenSlidingWindow` that gives old gen just after gc. should we use that here? it does a bunch of things to ensure we get the min value of old-gen-usage if a gc happened.

also nit: `promoted == 0` does not imply a gc.

Reply:
should be ok for rcas to consume each other, or we can do it in the cluster rca; or simply refactor

Reply:
You're right. I started factoring out a common RCA to do these computations, but I'm leaving it as a future iteration for now.